File: Official Windows Platform Sample/Windows 8 app samples/[C#]-Windows 8 app samples/C#/Windows 8 app samples/Push and periodic notifications client-side sample (Windows 8)/C#/PushNotificationsHelper/MaintenanceTask.cs
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//
// Copyright (c) Microsoft Corporation. All rights reserved

using System;
using System.Threading;
using System.Threading.Tasks;
using Windows.ApplicationModel.Background;
using Windows.Foundation;

namespace PushNotificationsHelper
{
    public sealed class MaintenanceTask : IBackgroundTask
    {
        public void Run(IBackgroundTaskInstance taskInstance)
        {
            Notifier notifier = new Notifier();

            // It's important not to block UI threads. Since this is a background task, we do need
            // to block on the channel operations completing
            notifier.RenewAllAsync(false).AsTask().Wait();
        }
    }
}

File: Official Windows Platform Sample/Windows Phone 8.1 samples/111541-Windows Phone 8.1 samples/Push and periodic notifications client-side sample/C++/PushNotificationsHelper.Shared/MaintenanceTask.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "MaintenanceTask.h"
#include "Helper.h"

using namespace PushNotificationsHelper;

using namespace Concurrency;
using namespace Windows::ApplicationModel::Background;

void MaintenanceTask::Run(IBackgroundTaskInstance^ taskInstance)
{
	auto notifier = ref new Notifier();

	// It's important not to block UI threads. Since this is a background task, we do need
    // to block on the channel operations completing
	create_task(notifier->RenewAllAsync(false)).wait();
}
